dev/Breadcrumb/BreadcrumbItem.cpp

```diff
@@ -0,0 +1,263 @@
+
+BreadcrumbItem::BreadcrumbItem()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_BreadcrumbItem);
+
+    SetDefaultStyleKey(this);
+}
+
+BreadcrumbItem::~BreadcrumbItem()
+{
+    RevokeListeners();
+}
+
+void BreadcrumbItem::RevokeListeners()
+{
+    m_breadcrumbItemButtonLoadedRevoker.revoke();
+}
+
+void BreadcrumbItem::OnApplyTemplate()
+{
+    __super::OnApplyTemplate();
+
+    winrt::IControlProtected controlProtected{ *this };
+
+    // TODO: Implement
+    m_breadcrumbItemButton.set(GetTemplateChildT<winrt::Button>(L"PART_BreadcrumbItemButton", controlProtected));
+
+    if (const auto& breadcrumbItemButton = m_breadcrumbItemButton.get())
+    {
+        m_breadcrumbItemButtonLoadedRevoker = breadcrumbItemButton.Loaded(winrt::auto_revoke, { this, &BreadcrumbItem::OnLoadedEvent });
+    }
+}
+
+void BreadcrumbItem::OnLoadedEvent(const winrt::IInspectable&, const winrt::RoutedEventArgs&)
+{
+    if (auto breadcrumbItemButton = m_breadcrumbItemButton.get())
+    {
+        m_breadcrumbItemButtonClickRevoker.revoke();
+        if (m_isEllipsisNode)
+        {
+            m_breadcrumbItemButtonClickRevoker = breadcrumbItemButton.Click(winrt::auto_revoke, { this, &BreadcrumbItem::OnEllipsisItemClick });
+        }
+        else
+        {
+            m_breadcrumbItemButtonClickRevoker = breadcrumbItemButton.Click(winrt::auto_revoke, { this, &BreadcrumbItem::OnBreadcrumbItemClick });
+        }
+    }
+
+    if (m_isEllipsisNode)
+    {
+        SetPropertiesForEllipsisNode();
+    }
+    else if (m_isLastNode)
+    {
+        SetPropertiesForLastNode();
+    }
+    else
+    {
+        ResetVisualProperties();
+    }
+}
+
+void BreadcrumbItem::SetItemsRepeater(const winrt::Breadcrumb& parent)
+{
+    m_parentBreadcrumb.set(parent);
+}
+
+void BreadcrumbItem::SetFlyoutDataTemplate(const winrt::IInspectable& newDataTemplate)
+{
+    if (auto const& dataTemplate = newDataTemplate.try_as<winrt::DataTemplate>())
+    {
+        m_ellipsisDataTemplate.set(dataTemplate);
+    }
+}
+
+void BreadcrumbItem::OnBreadcrumbItemClick(const winrt::IInspectable& sender, const winrt::RoutedEventArgs& args)
+{
+    if (const auto& breadcrumb = m_parentBreadcrumb.get())
+    {
+        auto breadcrumbImpl = winrt::get_self<Breadcrumb>(breadcrumb);
+        breadcrumbImpl->RaiseItemClickedEvent(Content());
+    }
+}
+
+void BreadcrumbItem::OnFlyoutElementPreparedEvent(winrt::ItemsRepeater sender, winrt::ItemsRepeaterElementPreparedEventArgs args)
+{
+    const auto& element = args.Element();
+
+    element.AddHandler(winrt::UIElement::PointerPressedEvent(),
+        winrt::box_value<winrt::PointerEventHandler>({this, &BreadcrumbItem::OnFlyoutElementClickEvent}),
+        true);
+    
+    element.PreviewKeyDown({ this, &BreadcrumbItem::OnFlyoutElementKeyDownEvent });    
+}
+
+void BreadcrumbItem::OnFlyoutElementKeyDownEvent(const winrt::IInspectable& sender, const winrt::KeyRoutedEventArgs& args)
+{
+    if (args.Key() == winrt::VirtualKey::Enter)
```

Comment:
Also when args.Key() == winrt::VirtualKey::Space?   You should ask Kat.

Reply:
that is defined in the spec as having the same functionality as the "Enter" key